websites/A/AnimeFever/presence.ts

```diff
@@ -0,0 +1,287 @@
+interface PageContext {
+  middleware: (ref: Window, ...args: unknown[]) => boolean;
+  exec: (
+    context: Presence,
+    data: PresenceData,
+    options?: { [key: string]: unknown }
+  ) => Promise<PresenceData> | PresenceData;
+}
+interface LocalizedStrings {
+  [key: string]: string;
+}
+interface AnimeVideoEntity {
+  "@type": string;
+  name: string;
+  duration: string;
+  description: string;
+  thumbnail: string;
+  thumbnailUrl: string;
+  contentUrl: string;
+  uploadDate: string;
+}
+
+interface AnimeEpisodeEntity {
+  "@type": string;
+  url: string;
+  name: string;
+  inLanguage: string;
+  subtitleLanguage: string;
+  episodeNumber: number;
+  thumbnailUrl: string;
+  position: number;
+}
+
+interface AnimeEntity {
+  "@id": string;
+  url: string;
+  "@type": string;
+  name: string;
+  alternateName: string;
+  description: string;
+  image: string;
+  inLanguage: string;
+  numberOfEpisodes: number;
+  aggregateRating: AggregateRating;
+}
+
+interface AggregateRating {
+  "@type": string;
+  ratingValue: string;
+  ratingCount: number;
+  worstRating: number;
+  bestRating: number;
+}
+
+interface ExecutionArguments {
+  strings: LocalizedStrings;
+  images: { [key: string]: string };
+  [key: string]: unknown;
+}
+
+function getQuery() {
+  const queryString = location.search.split("?", 2),
+    query =
+      queryString && queryString.length > 0 && queryString[1]
+        ? queryString[1].split("&").reduce(function (l, r) {
+            const entry = r ? r.split("=", 2) : null;
+            if (entry == null) return l;
+            return Object.assign(l, { [entry[0]]: entry[1] });
+          }, {})
+        : {};
+  return query;
+}
+function clamp(number: number, lower: number, upper: number) {
+  number = +number;
+  lower = +lower;
+  upper = +upper;
+  lower = lower === lower ? lower : 0;
+  upper = upper === upper ? upper : 0;
+  if (number === number) {
+    number = number <= upper ? number : upper;
+    number = number >= lower ? number : lower;
+  }
+  return number;
+}
```

Comment:
presence.getTimestamps() instead of this...

Reply:
does not work, XX:XX left still counting down after pausing.

Reply:
so just check if video is paused and delete endTimestamp then

Reply:
but the video is still active, deleting endTimestamp property is just unnecessary.

Reply:
> left still counting down after pausing.

Reply:
ye but i have the icon that tells you that its paused or playing...

Reply:
You can get from video paused state anyways so.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLVideoElement

Reply:
What, do you understand what im telling here?
You are arguing if removing the endTimestamp tells the user that the video is paused but im using an icon under smallImageKey to tell the state of the video...
So deleting the endTimestamp is unnecessary since the states are explained via smallImageKey property.
Either way my code is working and the provided getTimestamps method does not work correctly.

Reply:
Okey so, I literally tested it now, and everything works. Try do this:
```suggestion
```

and line 130 to:
```javascript
if (!videoInstance.paused)
            data.endTimestamp = presence.getTimestampsfromMedia(videoInstance )[1]
 ```
 and everything will work without this clamp thing :)

Reply:
![image](https://user-images.githubusercontent.com/9348108/112010508-abf6dd80-8b27-11eb-913c-39d81367e1ab.png)
![image](https://user-images.githubusercontent.com/9348108/112010547-b1542800-8b27-11eb-9c3e-5c8b612d7489.png)

Reply:
and i mean that you can do this without your function that you literally did. Not that user will see if video is paused.. bruh.. I meant that you can check before setting the endTimestamp if video is paused so it will not set 0

Reply:
asked & resolved in discord#presence-dev